contracts/oracles/yearn/YearnVaultMultiOracle.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastBytes32Bytes6.sol";
+import "@yield-protocol/utils-v2/contracts/token/IERC20Metadata.sol";
+
+import "@yield-protocol/vault-interfaces/IOracle.sol";
+
+import "./IYvToken.sol";
+
+/**
+ *@title  YearnVaultMultiOracle
+ *@notice Provides current values for Yearn Vault tokens (e.g. yvUSDC/USDC)
+ *@dev    Both peek() and get() are provided for convenience
+ *        Prices are calculated, never based on cached values
+ */
+contract YearnVaultMultiOracle is IOracle, AccessControl {
+    using CastBytes32Bytes6 for bytes32;
+
+    event SourceSet(
+        bytes6 indexed baseId,
+        bytes6 indexed quoteId,
+        address indexed source,
+        uint8 decimals
+    );
+
+    struct Source {
+        address source;
+        uint8 decimals;
+        bool inverse;
+    }
+
+    /**
+     *@notice This is a registry of baseId => quoteId => Source
+     *        used to look up the Yearn vault address needed to calculate share price
+     */
+    mapping(bytes6 => mapping(bytes6 => Source)) public sources;
+
+    /**
+     *@notice Set or reset an oracle source and its inverse
+     *@param  baseId id used for related base token (e.g. USDC)
+     *@param  vaultTokenId id for Yearn vault token (e.g. yvUSDC)
+     *@param  vaultToken address for Yearn vault token
+     */
+    function setSource(
+        bytes6 baseId,
+        bytes6 vaultTokenId,
```

Comment:
Is it necessary to require the yvToken to be a specific parameter? This could be `quoteId` and if you pass the yvToken as the first parameter everything would still work, right?

Reply:
yes good call